### PR TITLE
[RFC] Add a logout page as an alternative to the logout module

### PR DESCRIPTION
**Pull request by leofeyer**
This PR implements #236. @contao/developers Please review.

(The missing page icon will be added automatically when we merge the back end icons branch.)

### CODE REVIEW

**Comment by qzminski** [comment]
Just out of curiosity, why do you still use the legacy code for the new features?

**Comment by leofeyer** [comment]
How else would you add a new page type?

**Comment by qzminski** [comment]
I do not particularly mean the logic behind it but the coding style and not using the DIC.

**Comment by leofeyer** [comment]
Because it is legacy code. New legacy code :smile:

**Comment by DanielSchwiperich** [comment]
regarding legacy code, it maybe would a good idea to not use global namespace for new classes like also mentioned here https://github.com/contao/core-bundle/issues/451#issuecomment-190123917 by yourself that they are deprecated ;)

not sure if that was also considered *legacy code* by @qzminski

**Comment by qzminski** [comment]
The new legacy code works but is a short-term solution. When the time comes the ```tl_page``` class will have to be converted into ```Contao\DataContainer\PageContainer``` (or whatever you call it) and thus you will have to rework even more than just the old Contao 3 code.

As @DanielSchwiperich has pointed out the ```PageLogout``` could also be moved to the correct namespace already. No need for legacy here as it is a new feature with a new class. The page types registry of course has to remain as is in this PR because it's irrelevant here.

Why not do it now and just slightly adjust (e.g. dependencies) in the future if needed? The logic should not change after all, right? Same thing about CS and variable type-prefixes ```$objPage```.

File: src/Resources/contao/pages/PageLogout.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+class PageLogout extends \Frontend
+{
+
+	/**
+	 * Return a redirect response object
+	 *
+	 * @param PageModel $objPage
+	 *
+	 * @return Response
```
Comment (qzminski) [inline src/Resources/contao/pages/PageLogout.php]: I think it should be ```RedirectResponse``` to be precise.

**Comment by leofeyer** [comment]
Please accept the fact that we will be adding new features to the old code. Reworking the code is a long term goal and not every feature can be postponed until then.

> the ```PageLogout``` could also be moved to the correct namespace already.

If we knew today what the "correct namespace" was, I would do it. But right now, we don't even know how pages will be handled in the future, so there is no way of knowing the "correct namespace".

**Comment by leofeyer** [comment]
Merged in 6712cb9eaaf84135cd93b78702633d06ac566f67.

**Pull request merged**